NUnit 3 support

fstaffa opened the pull request:
* support for NUnit3, based on NUnit 2 support
* no compatibility layer for NUnit2, users will have to create the arguments from scratch
* tested all supported parameters and they work fine
* missing - automated tests, is there some way how such extension would be tested in Fake ?
* missing - do I need to test somehow that support for TeamCity [working directory](https://github.com/fsharp/FAKE/pull/1064/files#diff-c6cffee77df9960127aff31f4784d000R165) works as expected? I took the logic from NUnit client and I am not sure what is the expected behavior

inosik commented on src/app/FakeLib/UnitTest/NUnit/NUnit3.fs:
```diff
@@ -0,0 +1,234 @@
+      ToolPath : string
+
+      /// NUnit console runner name. ( `nunit3-console.exe`)
+      ToolName : string
```
Do we really need `ToolPath` & `ToolName`? Most (or all?) other helpers just have a `ToolPath` parameter.

fstaffa replied:
Makes sense, will merge these two.

inosik commented on src/app/FakeLib/UnitTest/NUnit/NUnit3.fs:
```diff
@@ -0,0 +1,234 @@
+      Domain : NUnit3DomainModel
+      
+      /// Allows you to specify the version of the runtime to be used in executing tests.
+      Framework : string
```
Would be nice if this would be a union type rather than `string`.

fstaffa replied:
I thought about it, but I am not sure about which values should be provided. NUnit documentation does not list these values as nicely as for NUnit3DomainModel or NUnit3ProcessModel. Instead, it accepts any value and checks at runtime that it is [available](https://github.com/nunit/nunit/blob/master/src/NUnitEngine/nunit.engine/RuntimeFramework.cs#L252). The best that can be done is listing some of them (e.g. mono, net-4.5, v4.0, 2.0, mono-4.0) and adding option for Other(string). Do you think this would be better?

forki commented:
Is this ready to merge?

fstaffa commented:
It is. I have tested it with all in this wrapper and it works fine.
@inosik proposed changing the argument of Framework from string to discriminated union but it probably won't bring much benefit.

forki commented:
ok then please change it now, so that we don't have breaking change later

fstaffa commented:
Ok, will add there common runtimes which are supported by NUnit 3 and we can expand them later as needed

fstaffa commented:
I have changed runtime to discriminated union. I have predefined .net frameworks (2.0, 3.5, 4.0, 4.5) and added Other for any other value.
I was not able to get it running (or get NUnit3 running at all) with mono runtime so I didn't predefine any mono runtime and it will have to be defined manually using Other.
Otherwise this is ready for merge.

inosik commented:
Maybe this might help you: https://github.com/nunit/nunit/blob/master/src/NUnitConsole/nunit3-console/ConsoleOptions.cs#L106
And this as well: https://github.com/nunit/nunit/blob/master/src/NUnitEngine/nunit.engine.tests/RuntimeFrameworkTests.cs#L219-L244

fstaffa commented:
Thank you, this looks good. I added them (except frameworks 1.0 and 1.1 which should not be supported by NUnit 3)

johnkors commented:
+1 on this feature. :)

forki commented:
thanks a lot
